runtime/context.go

```diff
@@ -137,3 +137,41 @@ func timeoutUnitToDuration(u uint8) (d time.Duration, ok bool) {
 	}
 	return
 }
+
+// isPermanentHTTPHeader checks whether hdr belongs to the list of
+// permenant request headers maintained by IANA.
+// http://www.iana.org/assignments/message-headers/message-headers.xml
+func isPermanentHTTPHeader(hdr string) bool {
+	if hdr != "" && hdr[0] == ':' {
```

Comment:
None, solid point. I was under the impression (could be wrong here) that gRPC reserved headers start with `:` and this was part of a check. Happy to remove it.